src/App.tsx

```diff
@@ -15,49 +17,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = 'NONE',
+  ALPHABET = 'ALPHABET',
+  LENGTH = 'LENGTH',
+}
+
+type SortOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: SortOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((prevItem, currentItem) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return prevItem.localeCompare(currentItem);
+      case SortType.LENGTH:
+        return prevItem.length - currentItem.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    return visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
```

Comment:
nit: Use a more semantic variable name, e.g., isSortingReversed instead of isReversed. This will improve code readability. 🎓